pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -643,6 +621,22 @@ func (s *DiscoveryServer) loadAssignmentsForClusterLegacy(push *model.PushContex
 	return l
 }
 
+// inSidecarScope checks if a given endpoint should be made available to a proxy
+func inSidecarScope(ep *model.IstioEndpoint, hostname model.Hostname, proxy *model.Proxy) bool {
+	if proxy == nil || proxy.SidecarScope == nil {
+		return true
+	}
+
+	// Any given hostname should only appear in one namespaces
+	var namespaceForHostname = make(map[model.Hostname]string)
+	for _, egress := range proxy.SidecarScope.EgressListeners {
```

Comment:
As a side note - can we call this field SidecarScope.Egress - there is no listening involved...

Reply:
But that's for separate PR - for now should this be cached ?